app/controllers/my_facilities_controller.rb

```diff
@@ -113,4 +92,13 @@ def force_cache?
   def report_with_exclusions?
     current_admin.feature_enabled?(:report_with_exclusions)
   end
+
+  def process_facility_stats(facilities, type)
+    stats_service = FacilityStatsService.new(accessible_facilities: @accessible_facilities, retain_facilities: facilities,
+                                             ending_period: @period, rate_numerator: type)
+    stats_service.call
+    @data_for_facility = stats_service.facilities_data
+    @stats_by_size = stats_service.stats_by_size
```

Comment:
If the FacilityStatsService just returns the size aggregation, then it will only have one return value and these pieces will be more clear.